core/shared/src/main/scala/zio/ZManaged.scala

```diff
@@ -664,6 +664,24 @@ final case class ZManaged[-R, +E, +A](reserve: ZIO[R, E, Reservation[R, E, A]])
   final def whenM[R1 <: R, E1 >: E](b: ZManaged[R1, E1, Boolean]): ZManaged[R1, E1, Unit] =
     ZManaged.whenM(b)(self)
 
+  /**
+   * Modifies this `ZManaged` to provide a canceler that can be used to eagerly
+   * execute the finalizer of this `ZManaged`. The canceler will run
+   * uninterruptibly, and if completed will cause the regular finalizer to not
+   * run.
+   */
+  final def withEarlyRelease: ZManaged[R, E, (URIO[R, _], A)] =
+    ZManaged[R, E, (URIO[R, _], A)] {
+      reserve.flatMap {
+        case Reservation(acquire, release) =>
+          Ref.make(true).map { finalize =>
+            val canceler  = (release(Exit.interrupt) *> finalize.set(false)).uninterruptible
```

Comment:
Oh right I forgot about release using  an `Exit[_, _]`. How about we do 2 variants - `withEarlyReleaseExit(Exit[_, _] => URIO)` and `withEarlyRelease`?

I was going back and forth on defaulting to `Exit.interrupt` but I'm convinced this is good. Let's just add a note about that on the scaladoc.